cmd/gitserver/server/server_grpc.go

```diff
@@ -484,6 +484,15 @@ func (gs *GRPCServer) IsPerforcePathCloneable(ctx context.Context, req *proto.Is
 	return &proto.IsPerforcePathCloneableResponse{}, nil
 }
 
+func (gs *GRPCServer) CheckPerforceCredentials(ctx context.Context, req *proto.CheckPerforceCredentialsRequest) (*proto.CheckPerforceCredentialsResponse, error) {
+	err := p4testWithTrust(ctx, req.P4Port, req.P4User, req.P4Passwd)
```

Comment:
```suggestion
	err := p4testWithTrust(ctx, req.GetP4Port(), req.GetP4User(), req.GetP4Passwd())
```

You should use the generated getters here.

Reply:
Oops, thanks I always keep mixing these. I assume the generated getters are because there might be some custom unmarshal logic? Or what's their purpose?

Reply:
It's to protect against unset fields, especially when you get nested messages, etc.

Otherwise, you'll need to sprinkle checks like `if req != nil`... everywhere. (Of course, if the behavior of your code should be different if the field is unset versus the default value ("") in this case, then you'll need to handle that yourself).